File: app/src/main/java/app/simple/buyer/util/AddItemRecyclerViewAdapter.java
package app.simple.buyer.util;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import app.simple.buyer.R;
import app.simple.buyer.entities.BuyItem;
import io.realm.OrderedRealmCollection;
import io.realm.RealmRecyclerViewAdapter;

/**
 * Created by Zakharovi on 22.01.2018.
 */

public class AddItemRecyclerViewAdapter extends RealmRecyclerViewAdapter<BuyItem, AddItemRecyclerViewAdapter.MyViewHolder> {

    public AddItemRecyclerViewAdapter(OrderedRealmCollection<BuyItem> data) {
        super(data, true);
        setHasStableIds(true);
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.cell_add_item, parent, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        final BuyItem obj = getItem(position);
        holder.buyItem = obj;
        //noinspection ConstantConditions
        holder.title.setText(obj.getName());
    }

    @Override
    public long getItemId(int index) {
        //noinspection ConstantConditions
        return getItem(index).getId();
    }

    class MyViewHolder extends RecyclerView.ViewHolder {
        TextView title;
        public BuyItem buyItem;

        MyViewHolder(View view) {
            super(view);
            title = view.findViewById(R.id.title);
//            deletedCheckBox = (CheckBox) view.findViewById(R.id.checkBox);
        }
    }
}

File: app/src/main/java/app/simple/buyer/util/AddItemRecyclerViewAdapter.kt
package app.simple.buyer.util

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import app.simple.buyer.R
import app.simple.buyer.entities.BuyItem
import io.realm.OrderedRealmCollection
import io.realm.RealmRecyclerViewAdapter

/**
 * Created by Zakharovi on 22.01.2018.
 */

class AddItemRecyclerViewAdapter(data: OrderedRealmCollection<BuyItem>) : RealmRecyclerViewAdapter<BuyItem, AddItemRecyclerViewAdapter.BuyItemViewHolder>(data, true) {
    init {
        setHasStableIds(true)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BuyItemViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.cell_add_item, parent, false)
        return BuyItemViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: BuyItemViewHolder, position: Int) {
        val obj = getItem(position)
        holder.buyItem = obj
        holder.title.text = obj?.name
    }

    override fun getItemId(index: Int): Long {
        return getItem(index)!!.id!!
    }

    inner class BuyItemViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var title: TextView = view.findViewById(R.id.title)
        var buyItem: BuyItem? = null
        init {
            //            deletedCheckBox = (CheckBox) view.findViewById(R.id.checkBox);
        }
    }
}
